source/dataset.py

```diff
@@ -13,11 +12,11 @@ def __init__(
         self,
         image_data: List[Image.Image],
         target: List[int],
-        label_encoder: LabelEncoder,
         transform: bool = None,
     ) -> None:
+        label_classes = {"cloudy": 0, "rain": 1, "shine": 2, "sunrise": 3}
         self.image_data = image_data
-        self.target = torch.LongTensor(label_encoder.transform(target))
+        self.target = torch.LongTensor([label_classes[t] for t in target])
```

Comment:
FYI (no need to change this), `torch.LongTensor` will be deprecated and `torch.tensor([...], dtype=torch.long)` is the recommended way to create tensors with specific data type.